Put credentials into separate file, not env, added additional cities …

desrod opened the pull request:
* Added credentials to separate file, secured by perms/apparmor as needed, not ENV vars
* Removed unused modules from imports
* Reformatted code with black and isort to match PEP
* Added additional countries to AMER and EMEA regions to fan-out cloned posts

desrod committed:
Put credentials into separate file, not env, added additional cities in AMER, EMEA, clean up formatting with black and isort.

dpb1 commented:
I got the following exception:

```
post-job.py  README.md
(.venv) dpb@helo:~/src/greenhouse$ ./post-job.py 1478151 --region americas emea
Traceback (most recent call last):
  File "./post-job.py", line 226, in <module>
    main()
  File "./post-job.py", line 221, in main
    next_page[0].click()
  File "/home/dpb/src/greenhouse/.venv/lib/python3.8/site-packages/selenium/webdriver/remote/webelement.py", line 80, in click
    self._execute(Command.CLICK_ELEMENT)
  File "/home/dpb/src/greenhouse/.venv/lib/python3.8/site-packages/selenium/webdriver/remote/webelement.py", line 633, in _execute
    return self._parent.execute(command, params)
  File "/home/dpb/src/greenhouse/.venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/home/dpb/src/greenhouse/.venv/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: element is not attached to the page document
  (Session info: chrome=89.0.4389.72)


```

(The script did still work, but all second page cities were not "published")

desrod committed:
Numerous fixes, expanded cities, read credentials from local json, support for sub-jobs in a single role, cleaned up with black/isort

desrod committed:
Updated README.md to reflect changes in _this_ forked version of the repository

desrod committed:
Adding additional cities to help increase velocity in finding candidates

desrod committed:
Added a filter for multiple roles in a single job req, updated README with more clarity

desrod committed:
Adding additional US/West cities, initial fixes for merged roles (not yet working)

desrod committed:
This is the beginning of a rework, closing a few bugs/issues. Added '--headless' option with 2FA support, also '--reset' for deleting posts prior to re-cloning, also fixed the multi-page update for missed cities, moving code around to separate functions, removed an unnecessary nested auth loop, and other performance tweaks. More to come.

desrod committed:
Oops, some more cleanup from the detritus left after running black, removed unused modules

desrod committed:
Added a quick fix for cloning a single post on a job with multiple, discrete posts

desrod committed:
More code refactor to support multiple posts in a single role, cleaning out unreachable code

hillpd committed:
Enhanced support for multiple posts within a single role

Fixed several bugs due to GH changing element names/structure.
Fixed marking posts live due to GH changing how tables get sorted.
Applied consistent sleep/wait conditions after each page update.
Disabled "--reset" until a fix can be addressed for multiple posts.
Reworked tool workflow so each post gets processed when no limit is passed.

```
$ ./post-job.py 1787567 --region americas --headless
Enter your 2FA token: 818970
[Harvesting job details]
-> Processing page 1
-> Processing page 2
-> Processing page 3
-> Processing page 4
[Creating posts for "Cloud Engineering - open source, remote"]
-> Processing americas
--> All locations already exist.
[Creating posts for "Linux Engineering - open source, remote"]
-> Processing americas
--> All locations already exist.
[Marking all job posts live]
-> Processing page 1
-> Processing page 2
-> Processing page 3
-> Processing page 4
All done! Now go bring those candidates through to offers!
```

hillpd committed:
Handle listings without page navigation elements

Fixed several bugs with single page listings.
Added support for testing Demo Jobs on the INTERNAL board.
Removed debug code that should not have been pushed.
